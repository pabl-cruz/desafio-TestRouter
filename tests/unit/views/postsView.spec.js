import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import PostsView from '@/views/PostsView.vue'



describe('PostsView', () => { 

    const wrapper = mount(PostsView)
    it('captura de contenido HTML en prueba y contrasta', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Probando la existencia del componente o vista PostsView ', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/posts',
                name: 'PostsViewVue',
                component: PostsView
            }],
        })
        router.push('/posts')
        await router.isReady()

        const wrapper = mount(PostsView, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapper.findComponent(PostsView).exists()).toBe(true)
    }) 
    

    const postsDummy = [
        { id: 1, name: 'Post 1' },
        { id: 2, name: 'Post 2' },
        { id: 3, name: 'Post 3' },
        { id: 4, name: 'Post 4' }
    ]

    test('renderiza lista de posts', async () => {
        const wrapper = mount(PostsView)

        //actualiza la variable de estado
        wrapper.vm.posts = postsDummy

        //actualiza DOM 
        await wrapper.vm.$nextTick()

        //revisar si se renderizan los LI
        expect(wrapper.findAll('li')).toHaveLength(4)
    })
 })

 