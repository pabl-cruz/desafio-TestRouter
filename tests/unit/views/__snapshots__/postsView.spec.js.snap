// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsView captura de contenido HTML en prueba y contrasta 1`] = `
<div class="posts">
  <h1>Vista de Posts</h1>
  <ul>
    <li>Post 1</li>
    <li>Post 2</li>
    <li>Post 3</li>
    <li>Post 4</li>
  </ul>
</div>
`;
